File: Iniciante/URI 1041.hs
main :: IO ()
main = do
    line <- getLine
    let (x:y:_) = map (read :: String -> Double) $ words line
    if x == 0 && y == 0 
        then putStrLn $ "Origem"
    else if x == 0 
        then putStrLn $ "Eixo Y"
    else if y == 0 
        then putStrLn $ "Eixo X"
    else if y > 0 && x > 0 
        then putStrLn $ "Q1"
    else if y > 0 && x < 0 
        then putStrLn $ "Q2"
    else if y < 0 && x < 0 
        then putStrLn $ "Q3"
    else if y < 0 && x > 0 
        then putStrLn $ "Q4"
    else
        return ()

File: Iniciante/URI 1041.php
<?php
    $line = explode(" ", trim(fgets(STDIN)));
    $x = floatval($line[0]);
    $y = floatval($line[1]);

    if ($x == 0 && $y == 0) {
        echo "Origem\n";
    } else if ($x == 0) {
        echo "Eixo Y\n";
    } else if ($y == 0) {
        echo "Eixo X\n";
    } else if ($y > 0 && $x > 0) {
        echo "Q1\n";
    } else if ($y > 0 && $x < 0) {
        echo "Q2\n";
    } else if ($y < 0 && $x < 0) {
        echo "Q3\n";
    } else if ($y < 0 && $x > 0) {
        echo "Q4\n";
    }
?>